normandy/recipes/tests/api/v2/test_api.py

```diff
@@ -159,6 +159,7 @@ def test_it_can_create_recipes(self, api_client):
 
         def test_it_can_create_recipes_with_only_filter_object(self, api_client):
             action = ActionFactory()
+            Channel.objects.get_or_create(slug="beta", name="Beta")
```

Comment:
First, we shouldn't `get_or_create` in tests, since that implies we aren't sure of the default state of the DB at the beginning of test, which isn't good.

Second, we should probably be using a Factory for this like we do other models, instead of creating objects by hand. One way that could help is to not think about the slug/name, and just do something like:

```python
channel = ChannelFactory()

res = api_client.post(
    "/api/v2/recipe/",
    {
        ...
        "filter_object": [{"type": "channel", "channels": [channel.slug]}],
    }
)
```

Reply:
Hmm... This isn't actually that nice. I appreciate the desire to be consistent, i.e. if recipes use factories, that we should do that for Country, Locale and Channel too. 
However, the code becomes more complex looking.

Before:
```python
def test_country_works(self, api_client):
    Country.objects.get_or_create(code="US", name="United States of America")
    Country.objects.get_or_create(code="DE", name="Germany")
    res = self.make_recipe(
        api_client, filter_object=[{"type": "country", "countries": ["US", "DE"]}]
    )
    assert res.status_code == 201, res.json()
    recipe_data = res.json()

    Recipe.objects.get(id=recipe_data["id"])
    assert recipe_data["filter_expression"] == (
        '(normandy.country in ["US","DE"]) && (true)'
    )
```

After:
```python
def test_country_works(self, api_client):
    country1 = CountryFactory()
    country2 = CountryFactory(code="DE")
    res = self.make_recipe(
        api_client,
        filter_object=[{"type": "country", "countries": [country1.code, country2.code]}],
    )
    assert res.status_code == 201, res.json()
    recipe_data = res.json()

    Recipe.objects.get(id=recipe_data["id"])
    assert recipe_data["filter_expression"] == (
        f'(normandy.country in ["{country1.code}","{country2.code}"]) && (true)'
    )
```

I think the code before *read* more easily. It's more natural and there's definitely less noise. Not to mention that now I have to use these weird looking variable names like `country2` and if I call it `country_de` that's almost worse. 

But I'm not going to kick up a fuzz. Just registering my observation.

Reply:
A lot of our factories by default use a random value for fields. If that were the csae here, then you wouldn't need to specify the code on the second factory, and could in fact do `country1, country2 = CountryFactory.build_batch(2)`. That doesn't get around having to refer to `country2.code` instead of just assuming "DE" thought.

It would also be fine to just hard code `CountryFactory(name="Germany", code="DE")` and then using "DE" explicitly throughout the rest of the test. Using these factories equivalently to the `get_or_create` you had before is also fine.